frappe/public/js/frappe/form/form.js

```diff
@@ -833,6 +833,18 @@ frappe.ui.form.Form = class FrappeForm {
 		}
 	}
 
+	discard(btn, callback, on_error) {
+		const me = this;
+		return new Promise((resolve) => {
+			// this.validate_form_action("Discard") // ?
```

Comment:
```suggestion
			// this.validate_form_action("Discard") // ?
```

Drop dead comments, validation not not really required here IMO. Validation on other action helps because they are saving some local changes. 


Even client script hooks I'd say aren't required for this.